starport/pkg/protoanalysis/protoanalysis.go

```diff
@@ -55,7 +55,45 @@ func ParseFile(path string) (File, error) {
 	return files[0], nil
 }
 
-// SearchRecursive recursively finds all proto files under path.
-func SearchRecursive(path string) ([]string, error) {
-	return localfs.Search(path, protoFilePattern)
+// CheckTypes check if the proto files into a folder contains a list of message names
+func CheckTypes(path string, names []string) error {
```

Comment:
Why do we need the variadic function? In this case, we pass an array to the method. if we move to a variadic function, we need to call this way:
```go
protoanalysis.HasMessages(ctx, protoPath, customFields...)
```

Reply:
I find them easier to use and when you only need to check a single entry, you don't even need to create a slice. I cannot see why not having it?

Reply:
If we return a boolean, how do we know the invalid custom field to show the error to the user? Because we are checking a list of fields, the user can pass one valid and another invalid. We must show an error message with the wrong field name. In this design, it's better to have a method to check one by one field name (`HasMessage(path, name string)`) instead check a list of names (`HasMessages(path string, names ...string)`)

Reply:
I thought it is fine to just return a generic error but yes, individual errors makes more sense. In this case can we make IsImported to only return an error?

Reply:
Yes, sure! with this change, we can also apply the variadic function suggestion because we can have a better error status

Reply:
done on

https://github.com/tendermint/starport/pull/1495/commits/e76c2d5692807342546cb38a1f1dcad56234a6bc